src/SFA.DAS.LevyTransferMatching.Web/Controllers/PledgesController.cs

```diff
@@ -142,14 +143,45 @@ public async Task<IActionResult> Location(LocationRequest request)
         [Route("create/location")]
         public async Task<IActionResult> Location(LocationPostRequest request)
         {
-            Dictionary<int, string> errors = await _orchestrator.ValidateLocations(request);
+            var multipleValidLocations = new Dictionary<int, IEnumerable<string>>();
+
+            var errors = await _orchestrator.ValidateLocations(request, multipleValidLocations);
+
             if (errors.Any())
             {
                 AddLocationErrorsToModelState(errors);
-                return RedirectToAction("Location", new { request.EncodedAccountId, request.AccountId, request.CacheKey });
+
+                return RedirectToAction(nameof(Location), new { request.EncodedAccountId, request.AccountId, request.CacheKey });
+            }
+
+            await _orchestrator.UpdateCacheItem(request);
+
+            if (multipleValidLocations.Any())
+            {
+                // Then surface a view to allow them to select the correct
+                // location, from a set of multiple valid locations
+                return RedirectToAction(nameof(LocationSelect), new { request.EncodedAccountId, request.CacheKey });
             }
 
+            return RedirectToAction("Create", new CreateRequest() { EncodedAccountId = request.EncodedAccountId, CacheKey = request.CacheKey });
+        }
+
+        [Authorize]
+        [Route("create/location/select")]
+        public async Task<IActionResult> LocationSelect(LocationSelectRequest request)
+        {
+            var viewModel = await _orchestrator.GetLocationSelectViewModel(request);
+
+            return View(viewModel);
+        }
+
+        [Authorize]
+        [Route("create/location/select")]
+        [HttpPost]
+        public async Task<IActionResult> LocationSelect(LocationSelectPostRequest request)
```

Comment:
I haven't isolated it completely, but it looks like if you post this page without disambiguating the second of 2 ambigious locations, there is no link in the validation partial to take the user to the error.

Reply:
Thanks for your help narrowing down the fix for this. It's now been implemented, and has been unit tested.